src-tauri/src/articulation_point.rs

```diff
@@ -0,0 +1,164 @@
+use defaultdict::DefaultHashMap;
+use petgraph::visit::IntoNeighbors;
+
+use crate::graph_p::Edge;
+use crate::graph_p::Graph;
+use crate::graph_p::Node;
+use core::cmp::min;
+
+pub fn articulation_point_helper(
+    graph: &Graph,
+    node_idx: String,
+    visited: &mut DefaultHashMap<String, bool>,
+    disc: &mut DefaultHashMap<String, i32>,
+    low: &mut DefaultHashMap<String, i32>,
+    time: &mut usize,
+    parent: String,
+    ap: &mut DefaultHashMap<String, bool>,
+) {
+    let mut children = 0;
+    visited.insert(node_idx.clone(), true);
+
+    *time += 1;
+    disc.insert(node_idx.clone(), *time as i32);
+    low.insert(node_idx.clone(), *time as i32);
+
+    let neighbors = graph.get_neighbors(node_idx.clone());
+    for neighbor in neighbors {
+        if !visited.get(&neighbor.name) {
+            children += 1;
+            articulation_point_helper(
+                &graph,
+                neighbor.name.clone(),
+                visited,
+                disc,
+                low,
+                time,
+                node_idx.clone(),
+                ap,
+            );
+            low.insert(
+                node_idx.clone(),
+                min(*low.get(&node_idx), *low.get(&neighbor.name)),
+            );
+
+            if !parent.eq("-1") && low.get(&neighbor.name) >= disc.get(&node_idx) {
+                ap.insert(node_idx.clone(), true);
+            }
+        } else if !neighbor.name.eq(&parent) {
+            low.insert(
+                node_idx.clone(),
+                min(*low.get(&node_idx), *disc.get(&neighbor.name)),
+            );
+        }
+    }
+
+    if parent.eq("-1") && children > 1 {
+        ap.insert(node_idx.clone(), true);
+    }
+}
+
+pub fn articulation_point(graph: Graph) -> Vec<petgraph::graph::NodeIndex> {
+    // hashmap
+    let mut disc = DefaultHashMap::<String, i32>::new();
+    let mut low = DefaultHashMap::<String, i32>::new();
+    let mut visited = DefaultHashMap::<String, bool>::new();
+    let mut ap = DefaultHashMap::<String, bool>::new();
+    let mut time = 0;
+    let parent = "-1".to_string();
+
+    for node in graph.get_nodes() {
+        if !visited.get(&node.name) {
+            articulation_point_helper(
+                &graph,
+                node.name,
+                &mut visited,
+                &mut disc,
+                &mut low,
+                &mut time,
+                parent.clone(),
+                &mut ap,
+            );
+        }
+    }
+
+    let mut articulation_points = Vec::new();
+    for key in ap.keys() {
+        if *ap.get(key) {
+            articulation_points.push(graph.get_node_idx(key.to_string()));
+        }
+    }
+    return articulation_points;
+}
+
+// Create a unit test for the Graph struct
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn initialize_graph() {
```

Comment:
Instead of having `G1` and `G2`, I would probably recommend breaking this test up into multiple tests. Also (nit), I would give the test a more descriptive name